### PR TITLE
Add freedesktop metadata

### PR DESCRIPTION
**Pull request by fpiesche**
This adds a new `dist` directory intended for files specific to distributions rather than actually part of the code or executable/bundle, containing the Linux desktop launcher (which currently lives in the `.github` directory for some reason) as well as FreeDesktop `metainfo.xml` and `releases.xml` files. The XML files contain metadata that can be used by Linux desktop environments to display information about the package - this is, for example, used by Flathub for their web listings.

The `releases.xml` file currently needs updating manually. I'm working on a GitHub action that can automatically update these files when a new GitHub release is made; once I've done this and tested it I'll submit a PR with a workflow to keep this file up to date along with GitHub releases :)

### CODE REVIEW

**Comment by fpiesche** [comment]
Should I move the AppImage build scripts into the `dist` folder as well, and the png icon to `src/images` where the other icons are?

**Comment by jardon** [comment]
> Should I move the AppImage build scripts into the `dist` folder as well, and the png icon to `src/images` where the other icons are?

im going to say confidently, *probably*

File: .github/linux-appimage-qt.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ chmod a+x linuxdeploy-plugin-checkrt-x86_64.sh
 
 cp -a "$GITHUB_WORKSPACE/build/translations" AppDir/usr/bin
 
-./linuxdeploy-x86_64.AppImage --appdir AppDir -d "$GITHUB_WORKSPACE"/.github/shadps4.desktop  -e "$GITHUB_WORKSPACE"/build/shadps4 -i "$GITHUB_WORKSPACE"/.github/shadps4.png --plugin qt 
+./linuxdeploy-x86_64.AppImage --appdir AppDir -d "$GITHUB_WORKSPACE"/dist/net.shadps4.shadps4.desktop  -e "$GITHUB_WORKSPACE"/build/shadps4 -i "$GITHUB_WORKSPACE"/.github/shadps4.png --plugin qt
 rm AppDir/usr/plugins/multimedia/libgstreamermediaplugin.so
 ./linuxdeploy-x86_64.AppImage --appdir AppDir --output appimage
 mv Shadps4-x86_64.AppImage Shadps4-qt.AppImage
```
Comment (jardon) [inline .github/linux-appimage-qt.sh]: the appimage name wont match here

File: .github/linux-appimage-sdl.sh [hunk 2]
```diff
@@ -17,5 +17,5 @@ chmod a+x linuxdeploy-plugin-checkrt-x86_64.sh
 # Build AppImage
 ./linuxdeploy-x86_64.AppImage --appdir AppDir
 ./linuxdeploy-plugin-checkrt-x86_64.sh --appdir AppDir
-./linuxdeploy-x86_64.AppImage --appdir AppDir -d "$GITHUB_WORKSPACE"/.github/shadps4.desktop  -e "$GITHUB_WORKSPACE"/build/shadps4 -i "$GITHUB_WORKSPACE"/.github/shadps4.png --output appimage
+./linuxdeploy-x86_64.AppImage --appdir AppDir -d "$GITHUB_WORKSPACE"/dist/net.shadps4.shadps4.desktop  -e "$GITHUB_WORKSPACE"/build/shadps4 -i "$GITHUB_WORKSPACE"/.github/shadps4.png --output appimage
 mv Shadps4-x86_64.AppImage Shadps4-sdl.AppImage
```
Comment (jardon) [inline .github/linux-appimage-sdl.sh]: the appimage name wont match here

**Review by jardon — CHANGES_REQUESTED** [review]
I think you should remove the change of the name field in the desktop shortcut and have that change be merged in this [PR](https://github.com/shadps4-emu/shadPS4/pull/1491)

**Comment by ngoquang2708** [comment]
Update ```REUSE.toml``` for newly added/moved files or else CI will fail.

**Review by jardon — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 3]
```diff
@@ -939,6 +939,9 @@ endif()
 install(TARGETS shadps4 BUNDLE DESTINATION .)
 
 if (ENABLE_QT_GUI AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    install(FILES ".github/shadps4.desktop" DESTINATION "share/applications")
-    install(FILES ".github/shadps4.png" DESTINATION "share/icons/hicolor/512x512/apps")
+    install(FILES "dist/net.shadps4.shadPS4.desktop" DESTINATION "share/applications")
+    install(FILES "dist/net.shadps4.shadPS4.releases.xml" DESTINATION "share/metainfo/releases")
+    install(FILES "dist/net.shadps4.shadPS4.metainfo.xml" DESTINATION "share/metainfo")
+    install(FILES ".github/shadps4.png" DESTINATION "share/icons/hicolor/512x512/apps" RENAME "net.shadps4.shadPS4.png")
+
```
Comment (ngoquang2708) [inline CMakeLists.txt]: Empty newline.

**Comment by ngoquang2708** [comment]
LGTM.

**Review by ngoquang2708 — APPROVED** [review]

**Comment by jardon** [comment]
@fpiesche can you resolve the conflicts so we can get this merged?

**Comment by fpiesche** [comment]
Ended up just exporting my changes to a patch, deleting my local branch, re-creating it and applying the patch to that. Apparently I've temporarily forgotten how to do a rebase without making a mess ._.

**Pull request merged**